### PR TITLE
feat(developer): eliminate source .model_info and .keyboard_info files 🎺

### PR DESCRIPTION
**Pull request by mcdurdin**
Removes all references to source .keyboard_info and .model_info files in the kmc compiler and types. Updates the model info compiler to support building purely from model sources.

Adds support for calculating `isRTL` and `license` fields in .model_info compiler.

Also:
* establishes @keymanapp/developer-utils shared module
* moves license validation into @keymanapp/developer-utils
* refactors kmc-model-info to a class and general cleanup.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc/src/commands/buildClasses/BuildProject.ts [hunk 1]
```diff
@@ -60,6 +66,10 @@ class ProjectBuilder {
   }
 
   async buildProjectTargets(activity: BuildActivity): Promise<boolean> {
+    if(activity.sourceExtension == KeymanFileTypes.Source.Project) {
+      return await this.buildTarget(this.project.projectFile, activity);
+    }
```
Comment (mcdurdin) [inline developer/src/kmc/src/commands/buildClasses/BuildProject.ts:71]: .keyboard_info and .model_info are built from the project

File: developer/src/kmc-model-info/src/model-info-compiler.ts [hunk 2]
```diff
@@ -1,163 +1,209 @@
...
 /* c8 ignore start */
-export class ModelInfoOptions {
+export class ModelInfoSources {
   /** The identifier for the model */
   model_id: string;
 
   /** The data from the .kps file, transformed to kmp.json */
   kmpJsonData: KmpJsonFile.KmpJsonFile;
 
   /** The path in the keymanapp/lexical-models repo where this model may be found (optional) */
```
Comment (darcywong00) [inline developer/src/kmc-model-info/src/model-info-compiler.ts:22]:
```suggestion
  /** The path in the keymanapp/lexical-models repo where this model may be found */
```

sourcePath looks required now?

File: resources/build/build-utils-ci.inc.sh [hunk 3]
```diff
@@ -141,9 +143,68 @@ function _builder_write_npm_version() {
...
+
+  # For each @keymanapp/ package, we'll do a local symlink, note that Windows
+  # mklink is internal to cmd!
+  for package in "${packages[@]}"; do
+    if [[ $package =~ ^@keymanapp/ ]]; then
+      # Creating local symlink under node_modules
+      local link_source=node_modules/$package
+
+      # lookup the link_target from top-level package.json/dependencies
+      local link_target="$(cat "$KEYMAN_ROOT/builder_package_publish.json" | jq -r .dependencies.\"$package\")"
```
Comment (darcywong00) [inline resources/build/build-utils-ci.inc.sh:171]: Do we want the platform agnostic call?
```suggestion
      local link_target="$(cat "$KEYMAN_ROOT/builder_package_publish.json" | "$JQ" -r .dependencies.\"$package\")"
```

File: web/src/test/manual/web/issue2924/source/dmg.dv.test/dmg.dv.test.kpj [hunk 4]
```diff
@@ -48,13 +48,6 @@
       <FileVersion></FileVersion>
       <FileType>.md</FileType>
     </File>
-    <File>
-      <ID>id_4b12eec2a4d0bfc7b5441a4a59691eab</ID>
-      <Filename>dmg.dv.test.model_info</Filename>
-      <Filepath>dmg.dv.test.model_info</Filepath>
-      <FileVersion></FileVersion>
-      <FileType>.model_info</FileType>
-    </File>
```
Comment (darcywong00) [inline web/src/test/manual/web/issue2924/source/dmg.dv.test/dmg.dv.test.kpj:57]: Should we also clean up .keyboard_info files in other web test folders?

web/srce/test/manual/web/issue917-context-and-notany/test_917/test_917.keyboard_info
web/srce/test/manual/web/issue2924/source/testvariable/testvariable.keyboard_info
web/srce/test/manual/web/issue3701/test3701/test3701.keyboard_info

File: developer/src/kmc-model-info/src/model-info-compiler.ts [hunk 5]
```diff
@@ -1,163 +1,209 @@
...
+    model_info.jsFileSize = this.callbacks.fileSize(sources.modelFileName);
+    if(model_info.jsFileSize === undefined) {
+      this.callbacks.reportMessage(ModelInfoCompilerMessages.Error_FileDoesNotExist({filename:sources.modelFileName}));
+      return null;
+    }
+
+    model_info.packageIncludes = sources.kmpJsonData.files.filter((e) => !!e.name.match(/.[ot]tf$/i)).length ? ['fonts'] : [];
+    model_info.version = sources.kmpJsonData.info.version.description;
+    model_info.minKeymanVersion = minKeymanVersion;
+    model_info.helpLink = HelpRoot + model_info.id;
```
Comment (darcywong00) [inline developer/src/kmc-model-info/src/model-info-compiler.ts:164]: Are we going to be publishing lexical-model help? I didn't see HelpRoot in the help.keyman.com repo
https://help.keyman.com/model

Comment (mcdurdin) [reply developer/src/kmc-model-info/src/model-info-compiler.ts:164]: This leaves the option open for the future, but no current plans :grin:

File: web/src/test/manual/web/issue2924/source/dmg.dv.test/dmg.dv.test.kpj (hunk 4 above)
Comment (mcdurdin) [reply web/src/test/manual/web/issue2924/source/dmg.dv.test/dmg.dv.test.kpj:57]: See #9551

**Review by jahorton — APPROVED** [review]
Partly LGTM, partly RSLGTM.

**Pull request merged**